src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c

```diff
@@ -9,67 +13,527 @@ void EnKarebaba_Destroy(Actor* thisx, GlobalContext* globalCtx);
     OBJECT_DEKUBABA,
     sizeof(EnKarebaba),
     (ActorFunc)EnKarebaba_Init,
     (ActorFunc)EnKarebaba_Destroy,
     (ActorFunc)EnKarebaba_Update,
-    (ActorFunc)EnKarebaba_Draw,
+    (ActorFunc)EnKarebaba_Draw
+};
+
+static ColliderCylinderInit sKarebabaHitbox = {
+    { 0xC, 0, 9, 0, 0x10, COLSHAPE_CYLINDER },
+    { 0, { 0x00000000, 0, 0 }, { ~0x00300000, 0, 0 }, 0, 1, 0 },
+    { 7, 25, 0, { 0, 0, 0 } }
+};
+
+static ColliderCylinderInit sKarebabaHitbox2 = {
+    { 0x0C, 0x11, 0, 0x39, 0x10, COLSHAPE_CYLINDER },
+    { 0, { ~0x00300000, 0, 8 }, { 0x00000000, 0, 0 }, 9, 0, 1 },
+    { 4, 25, 0, { 0, 0, 0 }}
+};
+
+static CollisionCheckInfoInit sKarebabaCheckInfo = {
+    1, 15, 80, 254
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(unk_4C, 0x9C4, ICHAIN_CONTINUE),
+    ICHAIN_U8(unk_1F, 1, ICHAIN_CONTINUE),
+    ICHAIN_S8(naviEnemyId, 9, ICHAIN_STOP)
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Karebaba/EnKarebaba_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Karebaba/EnKarebaba_Destroy.s")
+static Vec3f sVecZero1 = { 0.0f, 0.0f, 0.0f };
+
+static Color_RGBA8 sColorBlack = { 0, 0, 0, 0 };
+
+extern SkeletonHeader D_06002A40;
+extern AnimationHeader D_060002B8;
+
+// Display list for Deku Stick drop
+extern Gfx D_06003070[];
+
+// Display list for leaf base
+extern Gfx D_060010F0[];
+
+// Display list for mid third of stem
+extern Gfx D_06001330[];
+
+// Display list for lower third of stem
+extern Gfx D_06001628[];
+
+// Display list for upper third of stem
+extern Gfx D_06001828[];
+
+static Gfx* sDisplayLists[] = {
+    D_06001330,
+    D_06001628,
+    D_06001828
+};
+
+Vec3f sVecZero2 = { 0.0f, 0.0f, 0.0f };
+
+void EnKarebaba_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnKarebaba* this = THIS;
+
+    Actor_ProcessInitChain(this, sInitChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawFunc_Circle, 22.0f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_06002A40, &D_060002B8, &this->limbDrawTable, &this->transitionDrawTable, 8);
+    Collider_InitCylinder(globalCtx, &this->bodyHitbox);
+    Collider_SetCylinder(globalCtx, &this->bodyHitbox, this, &sKarebabaHitbox);
+    Collider_CylinderUpdate(this, &this->bodyHitbox);
+    Collider_InitCylinder(globalCtx, &this->headHitbox);
+    Collider_SetCylinder(globalCtx, &this->headHitbox, this, &sKarebabaHitbox2);
+    Collider_CylinderUpdate(this, &this->headHitbox);
+    func_80061ED4(&this->actor.colChkInfo, DamageTable_Get(1), &sKarebabaCheckInfo);
+
+    this->boundFloor = NULL;
+
+    if (this->actor.params == 0) {
+        EnKarebaba_SetupGrow(this);
+    } else {
+        EnKarebaba_SetupIdle(this);
+    }
+}
+
+void EnKarebaba_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnKarebaba* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->bodyHitbox);
+    Collider_DestroyCylinder(globalCtx, &this->headHitbox);
+}
+
+void EnKarebaba_DisableHitbox(EnKarebaba* this) {
```

Comment:
are you sure about this name? it makes sense in the code but the collider flags are so unknown we can't be sure they disable the collider?

also, use Collider instead of Hitbox

finally, disabling colliders is more generally handled by not subscribing to collision pools AC/AT/OC (ATtack, Attack Collider, Object Collider), cf `EnKarebaba_Update` and usage of `CollisionCheck_SetAT`, `CollisionCheck_SetAC`, `CollisionCheck_SetOC`

I'm really not sure this "disables" anything tbh

Reply:
Yeah, after commenting out the code, I would agree. I'll rename it to something more... generic.